ocs_ci/deployment/aws.py

```diff
@@ -134,18 +150,190 @@ def destroy_cluster(self, log_level="DEBUG"):
+                f'/aos-4_2/hosts/'
+            )
+
+        def deploy_prereq(self):
+            """
+            Overriding deploy_prereq from parent. Perform all necessary
+            prerequisites for AWSUPI here.
+            """
+            super(AWSUPI.OCPDeployment, self).deploy_prereq()
+
+            # setup necessary env variables
+            upi_env_vars = {
+                'INSTANCE_NAME_PREFIX': config.ENV_DATA['cluster_name'],
+                'AWS_REGION': config.ENV_DATA['region'],
+                'rhcos_ami': 'ami-06c85f9d106577272',
+                'route53_domain_name': config.ENV_DATA['base_domain'],
+                'vm_type_masters': config.ENV_DATA['master_instance_type'],
+                'vm_type_workers': config.ENV_DATA['worker_instance_type'],
+                'num_workers': "3"  # TODO: read in num_workers
+            }
+            for key, value in upi_env_vars.items():
+                os.environ[key] = value
+
+            # ensure environment variables have been set correctly
+            for key, value in upi_env_vars.items():
+                assert os.getenv(key) == value, f"{os.getenv(key)} != {value}"
+
+            # git clone repo from openshift-qe repo
+            clone_repo(
+                constants.OCP_QE_MISC_REPO, self.upi_repo_path
+            )
+
+            # create install-dir inside upi_script_path
+            relative_cluster_path = "../../../../../../"
+            os.symlink(
+                os.path.join(
+                    relative_cluster_path,
+                    os.path.basename(self.cluster_path.rstrip('/'))
+                ),
+                os.path.join(self.upi_script_path, "install-dir")
+            )
+
+            # NOT A CLEAN APPROACH: copy openshift-install and oc binary to
+            # script path because upi script expects it to be present in
+            # script dir
+            bindir = os.path.join(os.getcwd(), 'bin')
+            shutil.copy2(
+                os.path.join(bindir, 'openshift-install'),
+                self.upi_script_path,
+            )
+            shutil.copy2(
+                os.path.join(bindir, 'oc'), self.upi_script_path
+            )
+
+        def deploy(self, log_cli_level='DEBUG'):
+            """
+            Exact deployment will happen here
+
+            Args:
+                log_cli_level (str): openshift installer's log level
+                    (default: "DEBUG")
+            """
+            logger.info("Deploying OCP cluster")
+            logger.info(
+                f"Openshift-installer will be using loglevel:{log_cli_level}"
+            )
+
+            # Invoke upi_on_aws-install.sh
+            cidir = os.getcwd()
+            logger.info("Changing CWD")
+            try:
+                os.chdir(self.upi_script_path)
+            except OSError:
+                logger.exception(
+                    f"Failed to change CWD to {self.upi_script_path} "
```

Comment:
I think that this is problematic to change the CWD and leads why the oc is not visible as we assume its from ./bin of cwd I guess. Also maybe leads to problems with KUBECONFIG if cluster dir is set relatively?

Reply:
We are changing CWD because the script uses CWD to calculate different paths. We keep track of the initial CWD before we change, and are changing back to that original CWD after we execute the script.

Reply:
I see that it's needed by script but I giess it's the problem of why you don't have oc in yor path. So we should make this .bin dir exported to the path with its full path from  root /

Reply:
Two issues:
1. It's not only about oc - the script calculates other variables based on cwd such as the `working_dir` and `script_dir` which other variables depend on.
2. The PATH used by the shell script will be the PATH as it was when **python** was called. Even if we alter the path in our code, it won't apply to the PATH that the shell script will use. This means that `oc` needs to be on the path BEFORE we execute `run-ci`, as well as the other dependencies like `jq` and `aws`.